src/plus/integrations/authentication/gitlab.ts

```diff
@@ -1,18 +1,42 @@
 import type { AuthenticationSession, Disposable, QuickInputButton } from 'vscode';
 import { env, ThemeIcon, Uri, window } from 'vscode';
-import type { Container } from '../../../container';
-import type { HostingIntegrationId, SelfHostedIntegrationId } from '../providers/models';
-import type { IntegrationAuthenticationSessionDescriptor } from './integrationAuthentication';
-import { LocalIntegrationAuthenticationProvider } from './integrationAuthentication';
+import { HostingIntegrationId, SelfHostedIntegrationId } from '../providers/models';
+import type { IntegrationAuthenticationSessionDescriptor, StoredSession } from './integrationAuthentication';
+import {
+	CloudIntegrationAuthenticationProvider,
+	LocalIntegrationAuthenticationProvider,
+} from './integrationAuthentication';
 
 type GitLabId = HostingIntegrationId.GitLab | SelfHostedIntegrationId.GitLabSelfHosted;
 
-export class GitLabAuthenticationProvider extends LocalIntegrationAuthenticationProvider<GitLabId> {
-	constructor(
-		container: Container,
-		protected readonly authProviderId: GitLabId,
-	) {
-		super(container);
+export class GitLabAuthenticationProvider extends CloudIntegrationAuthenticationProvider<GitLabId> {
+	protected override get authProviderId(): GitLabId {
+		return HostingIntegrationId.GitLab;
+	}
+
+	protected override getCompletionInputTitle(): string {
+		return 'Connect to GitLab';
+	}
+	protected override async restoreSession({
+		sessionId,
+		ignoreErrors,
+	}: {
+		sessionId: string;
+		ignoreErrors: boolean;
+	}): Promise<StoredSession | undefined> {
+		const localSession = await this.readSecret(this.getLocalSecretKey(sessionId), ignoreErrors);
+		if (localSession != null) return localSession;
+
+		return super.restoreSession({
+			sessionId: sessionId,
+			ignoreErrors: ignoreErrors,
+		});
+	}
```

Comment:
Would you mind explaining why this override is needed? Seems like it's parent fn already checks the local key, and I thought we always wanted to check for the cloud key first.